docs/ops/matrix/Einsum_7.md

```diff
@@ -0,0 +1,248 @@
+where dimensions with labels `b` and `d` are reduced and the transpose is applied to get output layout `ca`.
+
+**NOTE**: *Einsum* operation can perform on the single operand and in this case the operation can transpose the input and reduce its dimensions.
+
+**NOTE**: Input ranks must be equal to a length of corresponding subscripts and dimensions to which the same labels in input subscripts 
+correspond must be equal in size.
+
+**NOTE**: A labels can repeat in the same input subscript, for example, `equation` equal to `aac,abd,ddde`. In this case the corresponding dimensions
+must match in size and the operand the operand will be replaced by its diagonal along these dimensions.
+For example, *Einsum* operation on the single 3D tensor of shape `[2, 4, 5, 4]` with `equation` equal to `ijkj->ij`.
+
+**NOTE**: The specification considers the primitive algorithm for *Einsum* operation for better understanding of the operation
+and does not recommend it for implementation.
+
+**NOTE**: The described algorithm can be improved by immediate dimension sum-reduction in the intermediate results if the corresponding labels absent in
+the input subscripts of subsequent inputs and the output subscript. It can significantly boosts performance and reduces memory costs.
+In the considered example, after the first step we can reduce a dimension corresponding to a label `d`.
+
+The output shape can be easily computed using the output subscript. It needs to concatenate sizes of dimensions to which labels in output subscript
+correspond in the specified order.
+
+Example 1 shows how *Einsum* computes inner product of two 1D tensors:
+
+```
+a1 = [1.0, 2.0, 3.0]
+a2 = [4.0, 5.0, 6.0]
+equation = "i,i->"
+output = 32.0
+```
+
+Example 2 shows how *Einsum* computes matrix-vector multiplication:
+
+```
+A = [[1.0, 2.0, 3.0],
+     [1.0, 2.0, 3.0]])
+b = [4.0, 5.0, 6.0]
+equation = "ij,j->i"
+output = [32.0, 32.0]
+```
+
+Example 3 shows how *Einsum* computes a trace for each batch object:
+
+```
+A = [[[1.0, 2.0, 3.0],
+      [4.0, 5.0, 6.0],
+      [7.0, 8.0, 9.0]],
+     [[2.0, 4.0, 6.0],
+      [8.0, 10.0, 12.0],
+      [14.0, 16.0, 18.0]],
+     ])
+equation = "kii->k"
+output = [15.0, 30.0]
+```
+
+Example 4 shows how *Einsum* extracts a diagonal for each batch object:
+
+```
+A = [[[1.0, 2.0, 3.0],
+      [4.0, 5.0, 6.0],
+      [7.0, 8.0, 9.0]],
+     [[2.0, 4.0, 6.0],
+      [8.0, 10.0, 12.0],
+      [14.0, 16.0, 18.0]],
+     ])
+equation = "kii->ki"
+output = [[1.0, 5.0, 9.0],
+          [2.0, 10.0, 18.0]]
+```
+
+Example 5 shows how *Einsum* transposes input tensor:
+
+```
+A = [[[1.0, 2.0, 3.0],
+      [4.0, 5.0, 6.0],
+      [7.0, 8.0, 9.0]]]
+equation = "ijk->kij"
+output = [[[1.0, 4.0, 7.0],
+           [2.0, 5.0, 8.0],
+           [3.0, 6.0, 9.0]]])
+```
+
+Aside a lower case label, ellipsis `...` can be used as a label in a subscript to cover broadcasted dimensions.
```

Comment:
Aside a lower case -> In addition to a lower case label, ...

Reply:
Fixed.